requirements.txt

```diff
@@ -1,7 +1,13 @@
-# Core deps.
-tensorflow~=2.13.0
-tensorflow-text~=2.13.0
-tensorflow-datasets
+# Tensorflow.
+tensorflow==2.14.0
+tensorflow-text==2.14.0.rc0
+
+# Torch.
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==2.0.1
```

Comment:
Do we really want to install torch+jax for all users as requirements?
Seems like pinning to exact versions and forcing installation as a requirement could bungle up people's environments unnecessarily

Reply:
I am a big fan of this doc... https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/

`install requires` and `pip install keras-nlp` -> minimal.
`requirements.txt` -> full dev environment (and optional! you don't need to use it). but we should use this for testing.

Could remove some of the pinning here though. The torch pin is from [keras-core](https://github.com/keras-team/keras-core/blob/main/requirements.txt) so just copied it. Jax pin is probably unnecessary.